File: Data Structures & Algorithms/anagram-groups/submission-0.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const result = new Map()

        for (const str of strs) {
            const count = new Array(26).fill(0)

            for (const char of str) {
                count[char.charCodeAt(0) - 'a'.charCodeAt(0)] ++;
            }
            const key = count.join(',');

            if (!result.has(key)) {
                result.set(key, [])
            }
            result.get(key).push(str)
        }
        return Array.from(result.values())
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int[] result = new int[arr.length];
        int greatestElement = arr[arr.length - 1];
        result[arr.length - 1] = -1;

        if (arr.length == 1) {
            return result;
        } 

        for (int i = arr.length - 2; i >= 0; i--) {
            result[i] = greatestElement;
            if (arr[i] > greatestElement) {
                greatestElement = arr[i];
            }
        }
        return result;
    }
}